tests/test_integrated_channels/test_api/test_sap_success_factors/test_views.py

```diff
@@ -91,20 +89,43 @@ def test_update(self, mock_current_request):
             'sapsf_company_id': 'test',
             'enterprise_customer': ENTERPRISE_ID,
             'sapsf_user_id': 893489,
-            'key': 'testing',
-            'secret': 'secret',
+            'encrypted_key': '',
+            'encrypted_secret': '',
             'user_type': 'user',
         }
         response = self.client.put(url, payload)
         self.sap_config.refresh_from_db()
         self.assertEqual(self.sap_config.sapsf_base_url, 'http://testing2')
         self.assertEqual(self.sap_config.sapsf_company_id, 'test')
-        self.assertEqual(self.sap_config.key, 'testing')
-        self.assertEqual(self.sap_config.secret, 'secret')
+        self.assertEqual(self.sap_config.decrypted_key, '')
+        self.assertEqual(self.sap_config.decrypted_secret, '')
         self.assertEqual(self.sap_config.user_type, 'user')
         self.assertEqual(self.sap_config.sapsf_user_id, '893489')
         self.assertEqual(response.status_code, 200)
 
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_populate_decrypted_fields(self, mock_current_request):
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+            context=self.enterprise_customer.uuid,
+        )
+        url = reverse('api:v1:sap_success_factors:configuration-detail', args=[self.sap_config.id])
+        client_secret = getattr(self.sap_config, 'secret', '')
+        payload = {
+            'sapsf_base_url': 'http://testing2',
+            'sapsf_company_id': 'test',
+            'enterprise_customer': ENTERPRISE_ID,
+            'sapsf_user_id': 893489,
+            'user_type': 'user',
+            'encrypted_secret': '1000',
+        }
+        self.client.put(url, payload)
+        self.sap_config.refresh_from_db()
+        self.assertEqual(self.sap_config.decrypted_secret, '1000')
+        populate_decrypted_fields_sap_success_factors(apps)
+        self.sap_config.refresh_from_db()
+        self.assertEqual(self.sap_config.encrypted_secret, client_secret)
```

Comment:
I don't think we will need this test anymore once this migration is done. Since this is a one-off change, I would suggest using a flag to skip coverage something like [this](https://github.com/openedx/edx-enterprise/blob/0e99db8b8fa39b3f38b9a4d23a8c2fa2c23dc0c3/integrated_channels/canvas/migrations/0037_canvasenterprisecustomerconfiguration_copy_id_and_secret_and_more.py#L6).

Reply:
@sameeramin i think if migration file is going to remain in code, we can keep the test case.